Adds core style overrides & bumps jetpack version

cpapazoglou opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

This is a follow up of https://github.com/Automattic/jetpack/pull/18097. Related issue is https://github.com/Automattic/jetpack/issues/18092

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Bumps jetpack version as a cache buster solution for styles loaded from wpcom (discussed here p1608286346170200-slack-CBG1CP4EN)
* Restructures core styles overrides to be compiled by a template
* Conditionally loads the correct styles override file
* Adds class to site-switcher and replaces `#adminmenuwrap > #adminmenu > .wp-first-item.menu-top-first > .wp-first-item.menu-top-first ` with `#adminmenuwrap > #adminmenu .site-switcher `
* Fixes Site-Switcher and Notes states

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

### **Simple Sites**
1) Enable nav-unification experiment (see paYJgx-1af-p2)
2) Apply the patch D54531
3) Change color-scheme from `https://wordpress.com/me/account` to a core color scheme (see list below)

### **Atomic Sites**
1) Select this branch on jetpack on an Atomic site (can't find it right now though. )
2) Install nav-unification enabler (needs to be proxied)
[nav-unification.zip](https://github.com/Automattic/jetpack/files/5717222/nav-unification.zip)
3) Change color-scheme from `/wp-admin/profile.php`

**Check the following color-schemes:**
- default
- light
-  Modern
-  Blue
-  Coffee
-  Ectoplasm
-  Midnight
-  Ocean
-  Sunrise

for the following:
**Masterbar:**
- [x] Masterbar background color
- [x] Masterbar hover color (Hover on Reader)
- [x] Masterbar active color (My Sites)

**Sidebar:**
- [x] Site card fadeout
- [x] Site card background color is not changing on hover
- [x] Site card icon color


#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
<!-- Guidelines: https://github.com/Automattic/jetpack/blob/master/docs/writing-a-good-changelog-entry.md -->
* Restructures color-scheme styles for nav-unification

Commit:
[not verified] Adds core style overrides & bumps jetpack version

frontdevde reviewed (commented):
First of all kudos for making the effort to switch the plain css over to generating the overrides per scheme via scss. This will significantly reduce the maintenance burden in the future while also making it less error prone.

> Select this branch on jetpack on an Atomic site (can’t find it right now though. )

Same here, the branch name doesn't show on the Jetpack beta plugin.

For Simple sites I applied the diff and noticed that it couldn't be cleanly applied which lead to the addition of  an `.orig` file on the diff. Seems be an svn issue. Let's make sure the php file is as expected on the diff and then remove the `.orig` file. 

While testing for Simple sites, the override was loaded in addition to the scheme as expected:

<img width="1321" alt="Screenshot 2020-12-21 at 08 24 06" src="https://user-images.githubusercontent.com/1562646/102751105-629bc400-4367-11eb-951e-5fd8486977ec.png">

On Simple sites, I noticed that the active state for the Notification bell in the wpcom masterbar isn't yet accounted for. Let's add that:

|Calypso|wp-admin|
|-|-|
|<img width="230" alt="Screenshot 2020-12-21 at 08 26 54" src="https://user-images.githubusercontent.com/1562646/102750692-8e6a7a00-4366-11eb-8d5f-09f0c239d3f8.png">|<img width="197" alt="Screenshot 2020-12-21 at 08 26 22" src="https://user-images.githubusercontent.com/1562646/102750704-94605b00-4366-11eb-907e-72067c3546a8.png">|

Edit: 

Same goes for the hover state on the Notification bell. 
See the relevant selectors here https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR73-R81

Commit:
[not verified] Adds notification icon hover state

Commit:
[not verified] Adds notification icon hover state, fix build

Commit:
[not verified] Cache Buster

Commit:
[not verified] Adds notes and rtl dependencies

frontdevde commented on modules/masterbar/admin-color-schemes/colors/_core-overrides.scss:
```diff
@@ -0,0 +1,48 @@
+/**
+ * Site Card
+ */
+ #adminmenu li.toplevel_page_site-card {
```
Looks like there's unnecessary whitespace at the beginning of the line here

cpapazoglou replied:
Removed 40d2b27

Commit:
[not verified] Adds active state

Commit:
[not verified] Removes whitespace

Commit:
[not verified] Adds important as last resort

cpapazoglou commented:
> First of all kudos for making the effort to switch the plain css over to generating the overrides per scheme via scss. This will significantly reduce the maintenance burden in the future while also making it less error prone.
> 
> > Select this branch on jetpack on an Atomic site (can’t find it right now though. )
> 
> Same here, the branch name doesn't show on the Jetpack beta plugin.
> 
> For Simple sites I applied the diff and noticed that it couldn't be cleanly applied which lead to the addition of an `.orig` file on the diff. Seems be an svn issue. Let's make sure the php file is as expected on the diff and then remove the `.orig` file.
> 
> While testing for Simple sites, the override was loaded in addition to the scheme as expected:
> 
> <img alt="Screenshot 2020-12-21 at 08 24 06" width="1321" src="https://user-images.githubusercontent.com/1562646/102751105-629bc400-4367-11eb-951e-5fd8486977ec.png">
> 
> On Simple sites, I noticed that the active state for the Notification bell in the wpcom masterbar isn't yet accounted for. Let's add that:
> 
> Calypso	wp-admin
> <img alt="Screenshot 2020-12-21 at 08 26 54" width="230" src="https://user-images.githubusercontent.com/1562646/102750692-8e6a7a00-4366-11eb-8d5f-09f0c239d3f8.png">	<img alt="Screenshot 2020-12-21 at 08 26 22" width="197" src="https://user-images.githubusercontent.com/1562646/102750704-94605b00-4366-11eb-907e-72067c3546a8.png">
> Edit:
> 
> Same goes for the hover state on the Notification bell.
> See the relevant selectors here https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR73-R81

Thanks @frontdevde for wrangling this testing! I have addressed your feedback!

frontdevde reviewed (commented):
I just noticed another very subtle issue when retesting with `blue`. The Site Switcher text has an alternative text color in nav unification that is not yet accounted for:

|Calypso|wp-admin|
|-|-|
|<img width="128" alt="Screenshot 2020-12-21 at 11 18 16" src="https://user-images.githubusercontent.com/1562646/102766447-7141a580-437e-11eb-81ba-125fa52f4e1b.png">|<img width="132" alt="Screenshot 2020-12-21 at 11 18 26" src="https://user-images.githubusercontent.com/1562646/102766462-756dc300-437e-11eb-9d46-c9f1c18a9cc9.png">|

For the color schemes originating in Calypso this was addressed here: 
https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR116-R130

Commit:
[not verified] Fixes site switcher colors

cpapazoglou commented on modules/masterbar/admin-color-schemes/colors/_overrides.scss:
```diff
@@ -114,14 +114,14 @@
 
 	// Site Card - Browse Sites button text & arrow
-	#adminmenuwrap > #adminmenu > .wp-first-item.menu-top-first > .wp-first-item.menu-top-first {
+	#adminmenuwrap > #adminmenu .site-switcher {
```
There are cases when "Browse Sites" is not rendered and thus not the first item

Commit:
[not verified] Fixes site card borders

Commit:
[not verified] Fixes site card borders color

frontdevde reviewed (commented):
Thanks for fixing the regressions as well! 🙌

Starting to look really solid! I've re-tested all color schemes again on Simple and the only issues I found were the following:

- On the Light color scheme the Notification bell  icon is white when it's black in Calypso.

- Related, on the Light color scheme the active state of the Notification bell is still off:

|Calypso|wp-admin|
|-|-|
|<img width="80" alt="Screenshot 2020-12-21 at 12 37 43" src="https://user-images.githubusercontent.com/1562646/102773970-60972c80-438a-11eb-8543-2828ec7db50d.png">|<img width="79" alt="Screenshot 2020-12-21 at 12 38 01" src="https://user-images.githubusercontent.com/1562646/102773981-62f98680-438a-11eb-9d10-8ebf7f29fe29.png">|

- In [Coffee, Blue, Ectoplasm, Midnight, Ocean, Sunrise] the Write button looks off on hover:

|Calypso|wp-admin|
|-|-|
|<img width="125" alt="Screenshot 2020-12-21 at 12 40 32" src="https://user-images.githubusercontent.com/1562646/102774072-8f150780-438a-11eb-8b84-dc91a9d9a88a.png">|<img width="99" alt="Screenshot 2020-12-21 at 12 40 23" src="https://user-images.githubusercontent.com/1562646/102774081-93d9bb80-438a-11eb-9b2a-c45bb0987006.png">|

Commit:
[not verified] Fixes notes icon on light

Commit:
[not verified] Fixes Write on hover

cpapazoglou commented:
Fixed both issues and refreshed the patch

frontdevde commented:
Looking good in all color schemes on Simple now 👏

Now that the branch is available in Jetpack Beta, I started testing as per your Atomic instructions and I'm noticing that the box-shadow at the bottom of the Masterbar differs:

|Calypso & Simple|Atomic|
|-|-|
|<img width="289" alt="Screenshot 2020-12-21 at 13 35 08" src="https://user-images.githubusercontent.com/1562646/102777871-c76c1400-4391-11eb-8886-3326447a4c97.png">|<img width="286" alt="Screenshot 2020-12-21 at 13 35 27" src="https://user-images.githubusercontent.com/1562646/102777910-d8b52080-4391-11eb-8deb-260f0c7e6a50.png">|

For the Calypso schemes this was addressed here https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR62-R65

Commit:
[not verified] Increases specificity

Commit:
Adds masterbar color

Commit:
Fixes tests

Commit:
Properly fixes test

frontdevde commented on jetpack.php:
```diff
@@ -16,7 +16,7 @@
 define( 'JETPACK__MINIMUM_WP_VERSION', '5.5' );
 define( 'JETPACK__MINIMUM_PHP_VERSION', '5.6' );
-define( 'JETPACK__VERSION', '9.3-alpha' );
+define( 'JETPACK__VERSION', '9.3-alpha-20201221' );
```
Was this for testing purposes or were you intending for this to go in?

cpapazoglou replied:
This was intentional in order to bust the caches, discussed here p1608286346170200-slack-CBG1CP4EN

frontdevde reviewed (approved):
Thanks for addressing all the feedback, looking slick now!

Tested as per testing instructions and can confirm it's now looking as expected in both Simple and WoA.

I'd appreciate if you could update the PR description with the latest changes so it provides some context for the next person reviewing. That said, I think we can have Crew take a look at this.

leogermani reviewed (approved):
@zinigor and I reviewed this PR and are approving it.

We found one issue in the WP-admin for atomic sites: when switching between color schemes, the master bar is not changing until a full page refresh.

This needs more testing and we aim to make it easier for testing approving this PR with a few usability issues still present.

slack discussion: p1608616675330200-slack-CBG1CP4EN

cpapazoglou commented:
Deployed r218681-wpcom
